compiler/stable_mir/src/mir/pretty.rs

```diff
@@ -69,6 +71,131 @@ pub fn pretty_statement(statement: &StatementKind) -> String {
     pretty
 }
 
+pub fn pretty_terminator(terminator: &TerminatorKind) -> String {
```

Comment:
Unrelated, I wonder if it's more efficient to treat these more like the `Display`/`Debug` traits, where they write directly to a formatter. That's unrelated, though.

Reply:
I'm just saying there could be some trait like

```
trait Pretty {
  fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Error;
}
```

Which duplicates the functionality of `Pretty`, possible even with a `context` passed as an argument, but which does the pretty-printing without having to allocate so many intermediate strings.

Reply:
Oh we also discussed that here https://github.com/rust-lang/rust/pull/117745#issuecomment-1810177226 I'm not sure if benefits outweigh the cons

Reply:
I had the same thought as @compiler-errors, but I think we can leave the implementation details to a later point when we can properly assess the trade-offs. The methods here will be helpful for users, and we can change them to invoke a `fmt` function to optimize the writing process.